src/dune_rules/exe_rules.ml

```diff
@@ -24,11 +25,20 @@ let linkages
     in
     let modes =
       if L.Map.mem exes.modes L.js
-      then Exe.Linkage.byte_for_jsoo :: modes
+      then (
+        match jsoo_compilation_mode with
+        | Js_of_ocaml.Compilation_mode.Whole_program -> Exe.Linkage.byte_for_jsoo :: modes
```

Comment:
Could we use type disambiguation to avoid the `Js_of_ocaml.Compilation_mode` prefix.
```diff
-  ~jsoo_compilation_mode
+  ~(jsoo_compilation_mode : Js_of_ocaml.Compilation_mode.t)
```

Reply:
Indeed. I have made the change.